src/analysis/retail/paladin/holy/modules/dragonflight/tier/T30TierSet.tsx

```diff
@@ -0,0 +1,136 @@
+import SPELLS from 'common/SPELLS';
+import { TIERS } from 'game/TIERS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { calculateEffectiveHealing, calculateOverhealing } from 'parser/core/EventCalculateLib';
+import Events, { HealEvent, ResourceChangeEvent } from 'parser/core/Events';
+import BoringValueText from 'parser/ui/BoringValueText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { TALENTS_PALADIN } from 'common/TALENTS';
+import { formatNumber } from 'common/format';
+import HIT_TYPES from 'game/HIT_TYPES';
+import { getLightsHammerHeals } from '../../../normalizers/CastLinkNormalizer';
+import { SpellLink } from 'interface';
+
+const CRIT_HEAL_AMOUNT = 1.65;
+const TWO_PIECE_HEAL_INC = CRIT_HEAL_AMOUNT / 1.5; // 210% heal instead of 150% on crit
+const HOLY_PRISM_INC = 0.8;
+
+class T30HpalTierSet extends Analyzer {
+  has4Piece: boolean = false;
+  hsHealing: number = 0;
+  hsOverhealing: number = 0;
+  hpGained: number = 0;
+  hpWasted: number = 0;
+  lhTiktok: boolean = false; // if true, count set of lh heals
+  fourPcHealing: number = 0;
+  fourPcOverhealing: number = 0;
+  countedHeals: Set<HealEvent> = new Set();
+  extraLhHits: number = 0;
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.T30);
+    if (!this.active) {
+      return;
+    }
+    this.has4Piece = this.selectedCombatant.has4PieceByTier(TIERS.T30);
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.HOLY_SHOCK_HEAL),
+      this.onHolyShockHeal,
+    );
+    this.addEventListener(
```

Comment:
maybe needs a `if (!this.has4Piece) { return }` here, or checks in the event handlers?